lib/linguist/heuristics.yml

```
@@ -709,6 +709,10 @@ disambiguations:
   - language: Markdown
     # Markdown syntax for scdoc
     pattern: '^#+\s+(NAME|SYNOPSIS|DESCRIPTION)'
+- extensions: ['.script']
+  rules:
+  - language: URScript
+    pattern: '^\s*def\s*[a-zA-Z_][a-zA-Z0-9_]*\s*\([^)]*\)\s*:[\s\S]*\send($|\s)'
```

This looks much better, thanks. I think there's still room for improvement:

```suggestion
    pattern: '^\s*def\s+[a-zA-Z_]\w*\s*\([^)]*\)\s*:[\s\S]*\send\b'
```

- `^\s*def\s*` replaced with `^\s*def\s+` as it appears at least one space is required
- `[a-zA-Z0-9_]*` is the same as `\w*`
- `\s*end($|\s)` is simplified to `\s*end\b` as `\b` matches a word boundary, which is equivalent to the end of the word `end`.

@Alhadis may have some further suggestions.

---

1.	`\send\b` will match a (presumably illegal) construction like `end.`, `end-`, `end(`, and many, _many_ others. 
2.	For performance's sake, `[\s\S]*` should match conservatively (`[\s\S]*?`) so that anything past the first `\send\b` isn't searched and included as part of the match:

	~~~ruby
	#!/usr/bin/env ruby
	greedy  = /^\s*def\s*[a-zA-Z_][a-zA-Z0-9_]*\s*\([^)]*\)\s*:[\s\S]*\send($|\s)/
	thrifty = /^\s*def\s*[a-zA-Z_][a-zA-Z0-9_]*\s*\([^)]*\)\s*:[\s\S]*?\send($|\s)/
	input   = File.read("samples/URScript/admittance_control.script")

	greedy  =~ input; $&.length # => 10019
	thrifty =~ input; $&.length # => 126
	~~~